src/components/Nav/Nav.js

```diff
@@ -1,7 +1,182 @@
-import React from 'react';
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { SIDEBAR_INSIDE } from './SIDEBAR_INSIDE';
+import SideInside from './SideInside';
 
 const Nav = () => {
-  return <div>hi</div>;
+  const [displayShow, setDisplayShow] = useState(false);
+
+  const showSidebar = () => {
+    setDisplayShow(!displayShow);
+  };
+
+  return (
+    <>
+      <Navigation>
+        <NavContainer>
+          <NavLeft>
+            <NavBt onClick={showSidebar}>
+              <MenuImg src="images/nav/menu.png" alt="menu" />
+            </NavBt>
+            <Logo src="images/nav/needed.png" alt="log" />
+          </NavLeft>
+          <TextBox>
+            <Menu>채용</Menu>
+            <Menu>이벤트</Menu>
+            <Menu>직군별 연봉</Menu>
+            <Menu>이력서</Menu>
+            <Menu>커뮤니티</Menu>
+            <Menu>프리랜서</Menu>
+            <Menu>AI 합격예측</Menu>
+          </TextBox>
+          <NavRight>
+            <SearchWrap>
+              <Search src="images/nav/search.png" alt="search" />
+            </SearchWrap>
+            <SignUpBt>회원가입/로그인</SignUpBt>
+            <RoundBt>기업서비스</RoundBt>
+          </NavRight>
+        </NavContainer>
+      </Navigation>
+      <SlideWrap>
+        <SlideBox displayShow={displayShow}>
+          {SIDEBAR_INSIDE.map(list => {
+            return (
+              <SideInside
+                key={list.index}
+                id={list.id}
+                title={list.title}
+                url={list.url}
+                subNav={list.subNav}
+              />
+            );
+          })}
```

Comment:
```suggestion
          {SIDEBAR_INSIDE.map((list,index) => {
            return (
              <SideInside
                key={index}
                {...list}
              />
            );
          })}
```

- index값은 SIDEBAR_INSIDE 데이터 안에 없는것 같네요 index는 map의 callback 함수의 두번째 인자로 전달됩니다.


- 객체안에 props로 전달될 속성들이 모두 들어있고 key값과 props로 전달할 이름이 동일하다면 spread 연산자 활용해서 위와 같이 표현할수도 있습니다 그러면 `key={value}` 형태로 객체의 모든 프로퍼티가 props로 전달됩니다.

Reply:
key 값과 props로 전달할 이름이 동일하면 이렇게 간단하게 표현할 수 있는거였군요!! map은 알면알수록 어렵지만 재밌는 함수같아요

Reply:
JSX에서 제공해주는 조금 더 편리한 문법이라서 필요한 상황에서 적절히 잘 사용하시면 좋습니다 👍